### PR TITLE
detailed error handling for pdfcomponent

### PR DESCRIPTION
**Pull request by Ribhav-Jain22**
…nd the file where the error occurred.

# Pull Request Template

## Description

created a helper function in helper/PdfHelper.js that logs the error message, stack trace, and the file where the error occurred

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which improves code readability and/or UI Improvement)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Yes
- [x] No

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have fixed SonarCloud errors
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: vue-project/src/helpers/PdfHelper.js [hunk 1]
```diff
@@ -1,15 +1,23 @@
+// src/helpers/PdfHelper.js
+
 import { _GSPS2PDF } from '../ghostscript-utils';
+import { printErrorOnConsole } from './Error/PrintError';
 
 function compressPDF(pdf, filename, compressionLevel, onLoadComplete, onProgress, onStatusUpdate) {
```
Comment (ad-mittal) [inline vue-project/src/helpers/PdfHelper.js:10]: add data types

File: vue-project/src/ghostscript-utils.js [hunk 2]
```diff
@@ -7,7 +10,7 @@ export function _GSPS2PDF(dataStruct, responseCallback, progressCallback, status
 
         let gsArguments;
         switch (compressionLevel) {
-            case 'HIGH':
+            case compressionLevel.HIGH:
```
Comment (ad-mittal) [inline vue-project/src/ghostscript-utils.js]: Check casing

File: vue-project/src/components/PdfConverter.vue [hunk 3]
```diff
@@ -98,7 +99,7 @@ export default {
         try {
           await this.beginCompression(url, filename, this.compressionLevel);
         } catch (error) {
-          this.handleError('An error occurred during compression. Please try again.', error);
+          printErrorOnConsole(error, 'pdfconverter.vue');
```
Comment (ad-mittal) [inline vue-project/src/components/PdfConverter.vue]: remove filename, use logger function to retrieve filename

**Review by Ribhav-Jain22 — COMMENTED** [review]
all the changes are concerning pdfcomponent  and can be mergerd

**Review by ad-mittal — APPROVED** [review]

**Pull request merged**